hacker_Rank/May4_Answer.py

```diff
@@ -0,0 +1,31 @@
+def array_rotation(a, k):
+    result = []
+    print(a)
+    step = 0
+    for i in range(k):
+        print("ref", a)
+        print(i)
+        temp = a[0]
+        a[0] = a[(0+1)]
+        a[(0+1)] = temp
+        print("fist_loop:", a)
+        step = 0
+
+        while step < len(a)-2:
+            # print(a)
+            step_temp = a[0]
+            a[0] = a[step + 2]
+            a[step + 2] = step_temp
+            step += 1
```

Comment:
can you explain me this logic ?

Reply:
ex: a= [1,2,3,4,5]
11 . first loop output i get = a= [2,1,3,4,5] 
so to rearrange rest of all the variables in the respective position i need to do swap the number from a[0] to a[step+2] so i need to increment the step every time so that each value at a[0] to will swap to 1step forward

11. first loop output = a=[2,1,3,4,5]
in while loop 
for step 0  : a=[3,1,2,4,5]
for step 1 :  a=[4,1,2,3,5]
for step 2 : a =[5,1,2,3,4]

so finally here i have moved all the variable 1 step forward that is when k=1
so this list will go to for loop again if k >1 it will print the same list at line22